tests/unit/s2n_io_test.c

```diff
@@ -102,6 +102,8 @@ int main(int argc, char **argv)
                         s2n_test_real_interrupt(io_pair.client, n_times, &counter));
                 EXPECT_EQUAL(result, S2N_TEST_SUCCESS);
                 EXPECT_EQUAL(counter, n_times);
+                /* Child process exit don't close io_pair properly, so manually close it*/
+                EXPECT_SUCCESS(s2n_io_pair_close(&io_pair));
```

Comment:
Would this work if the io_pair was created in a scope that ends before the exit call? Something like this maybe?
```
            {
                DEFER_CLEANUP(struct s2n_test_io_pair io_pair = { 0 }, s2n_io_pair_close);
                EXPECT_SUCCESS(s2n_io_pair_init(&io_pair));

                uint8_t counter = 0;
                int result = 0;
                S2N_IO_RETRY_EINTR(result,
                        s2n_test_real_interrupt(io_pair.client, n_times, &counter));
                EXPECT_EQUAL(result, S2N_TEST_SUCCESS);
                EXPECT_EQUAL(counter, n_times);
            }
            exit(0);
```